src/controllers/auth/resetPassword.ts

```diff
@@ -1,26 +1,107 @@
+        return true
+      }
+      const validPassword = await req.user.comparePassword(value)
+
+      if (!validPassword) {
+        throw new Error('Current Password must match the saved password.')
+      }
+
+      return true
+    })
+    .run(req)
+  await check('password', 'Password must be at least 8 characters long.').isLength({ min: 8 }).run(req)
+  await check('passwordConfirmation', 'Passwords do not match.').equals(req.body.password).run(req)
 
   const errors = validationResult(req)
 
   if (!errors.isEmpty()) {
     return res.status(422).json({ errors: errors.array() })
   }
-  return res.status(501).send()
+
+  try {
+    await User.update(req.user, { password: await hash(req.body.password, 10) })
+    return res.status(204).send()
+  } catch (error) {
+    logger.error(`Unable to update password for user with id: ${req.user.id}`)
+    return res.status(400).json({ error: 'User not found' })
+  }
 }
 
-export const postReset = async (req: Request, res: Response): Promise<Response> => {
-  await check('password', 'Password must be at least 4 characters long.').isLength({ min: 8 }).run(req)
+// Guest reset password
+// /password/reset
+export const postGuestResetPassword = async (req: Request, res: Response): Promise<Response> => {
+  await check('password', 'Password must be at least 8 characters long.').isLength({ min: 8 }).run(req)
+  await check('userId', 'The userId must be an int.').isInt().run(req)
   await check('passwordConfirmation', 'Passwords must match.').equals(req.body.password).run(req)
 
   const errors = validationResult(req)
 
   if (!errors.isEmpty()) {
     return res.status(422).json({ errors: errors.array() })
   }
-  return res.status(501).json({ error: 'not implemented' })
+
+  const token = req.body.token
+  const userId = req.body.userId
+  let user: User
+
+  try {
+    user = await User.findOneOrFail(userId)
+  } catch (error) {
+    logger.error(error)
+    return res.status(404).json({ error: 'User not found for the given userId' })
+  }
+
+  let passwordReset: PasswordReset
+
+  try {
+    passwordReset = await PasswordReset.findOneOrFail({ where: { user } })
+  } catch (error) {
+    logger.error(error)
+    return res.status(404).json({ error: 'Token not found for the given email address' })
+  }
+
+  const expiration = DateTime.fromJSDate(passwordReset.createdAt, { zone: 'America/Chicago' })
+    .toUTC()
+    .plus(Duration.fromObject({ minutes: 60 }))
+
+  if (DateTime.utc().toMillis() >= expiration.toMillis()) {
+    return res.status(422).send({ error: 'This password reset token is invalid.' })
+  }
+
+  // Compare token
+  // const isTokenValid = await compare(token, passwordReset.token)
+  const isTokenValid = token === passwordReset.token
+
+  if (!isTokenValid) {
+    return res.status(500).json({ error: 'Invalid token' })
+  }
+
+  await PasswordReset.delete(passwordReset)
+  await User.update(user, { password: await hash(req.body.password, 10) })
```

Comment:
We could place a try/catch around both of these lines and return `something went wrong` in the catch:
```suggestion
  try {
    await PasswordReset.delete(passwordReset)
    await User.update(user, { password: await hash(req.body.password, 10) })
  } catch(error) {
    logger.error(error)
    return res.status(500).json({ error: 'something went wrong' })
  }
```

Reply:
Fixed